wordpress-mobile/WordPress-Android
# Address AT issues after domain registration

Fixes #10100

This PR can be reviewed right now, but merged after [Flux-C counterpart](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1298) is merged, and correct commit has is reflected in this branch.

Update:
After more testing with more than 50 domains, I realized that there is not 100% sure way to do AT after fresh domain registration without some sort of problem. Most common ones are:

- Primary domain switches to "wpcomstaging".
- AT get's stuck even on empty site.
- AT errors out for some reason.

Additional issue: when you install a plugin during AT there is some sort of server-side delay, between completion of AT and when the plugin is marked as installed. This often times happen when you perform AT on freshly registered domains (

PR adds a delay/retry logic for domain registration and fetching plugin details.

In addition, this PR removes the "Congratulations" screen after the successful domain registration during AT. Instead, we immediately prompt used to start Automated Transfer after the domain is registered.

Sometimes domain is not ready for AT, so we notify used about this.
Sometimes everything works!

To test:

1. On a Simple site without a Custom Domain and with a Domain Credit, perform an AT by installing a plugin and registering a domain.
2. When domain is registered, notice that AT prompt is visible.
3. Installing a plugin will either work and you will see an error message, that site is not yet ready.
4. When AT is complete, notice that "INSTALL" button of plugin changes to "INSTALLED".

PS: `viewDomainRegistrationActivity` in `ActivityLauncher` is left for compatibility with `develop` branch.

Update release notes:
